app/README.md

```diff
@@ -0,0 +1,246 @@
+
+```javascript
+class ViewTransactionChargeCommonService {
+  static async go (billRun, transactionId) {
+    let transaction = await this._transaction(transactionId)
+
+    return this._response(transaction)
+  }
+
+  static _transaction (transactionId) {
+    return TransactionModel.query().findById(transactionId)
+  }
+
+  static _response (invoice) {
+    throw Error("Extending service must implement '_response()'")
+  }
+}
+```
+
+Note that the common `_response()` method will throw an error to ensure it is not inadvertently used. The ruleset based services can then undergo a final change. Instead of duplicating functionality they extend the common service and only implement their own version of `_response()`.
+
+```javascript
+class ViewTransactionChargeSrocService extends ViewTransactionChargeCommonService {
+  static _response (transaction) {
+    const presenter = new ViewTransactionChargeSrocPresenter(transaction)
+
+    return presenter.go()
+  }
+}
+
+# ...
+
+class ViewTransactionChargePresrocService extends ViewTransactionChargeCommonService {
+  static _response (transaction) {
+    const presenter = new ViewTransactionChargePresrocPresenter(transaction)
+
+    return presenter.go()
+  }
+}
+```
+
+### Translators and Presenters
+
+Continuing our example we'll focus on presenters but refactoring of translators and presenters across rulesets will be the same.
+
+The key difference from services is that there is no 'main' presenter. The determination of what the ruleset is and therefore which class types need to be used is made just once in the 'main' service. The ruleset services which just reference the ruleset presenters directly.
+
+Imagine the `ViewTransactionChargePresenter` before refactoring looks like this.
+
+```javascript
+class ViewTransactionChargePresenter extends BasePresenter {
+  _presentation (data) {
+    return {
+      id: data.id,
+      billableDays: data.regimeValue4,
+      abstractableDays: data.regimeValue5,
+      volume: data.lineAttr5,
+      source: data.regimeValue6,
+      season: data.regimeValue7,
+      loss: data.regimeValue8,
+      secondPartCharge: data.regimeValue16,
+      compensationCharge: data.regimeValue17,
+      eiucSource: data.regimeValue13,
+      waterUndertaker: data.regimeValue14,
+      region: data.regimeValue15,
+      s127Agreement: data.regimeValue12,
+      s130Agreement: data.regimeValue9,
+      abatementAdjustment: data.regimeValue11
+    }
+  }
+}
+```
+
+The existing presenter should be renamed using `git mv app/presenters/view_transaction_charge.presenter.js app/presenters/view_transaction_charge_presroc.presenter.js`. This should then be copied as `app/presenters/view_transaction_charge_sroc.presenter.js`. Finally, the new `ViewTransactionChargeSrocPresenter` should be updated based on whatever changes are needed for the new ruleset.
```

Comment:
Might be a little clearer to put the rename command on its own line in a block? Also is there a specific reason for renaming with a git command vs renaming then adding?

Reply:
Agreed and updated!

Regards using `git mv` vs just doing it manually. That will work in most cases but you are depending on git being clever enough to figure out that is what happened. Using `git mv` makes the change explicit in your git history that this was a rename and not a file getting deleted and a new one added.

Reply:
Ah I see! Yes that makes sense, thanks.